src/components/Common/UI/ComboBox/ComboBox.tsx

```diff
@@ -0,0 +1,102 @@
+import type { Key } from 'react-aria-components'
+import {
+  ComboBox as AriaComboBox,
+  Button,
+  Input,
+  ListBox,
+  ListBoxItem,
+  Popover,
+} from 'react-aria-components'
+import { useTranslation } from 'react-i18next'
+import type { FocusEvent, InputHTMLAttributes } from 'react'
+import { useId } from 'react'
+import classNames from 'classnames'
+import type { FieldLayoutProps } from '../FieldLayout'
```

Comment:
You'll also want to update this to use `SharedFieldLayoutProps`

```suggestion
import type { SharedFieldLayoutProps } from '../FieldLayout'
```

that'll restrict it to only the props that are meant to be shared across input components

Reply:
Updated - WIll be in next revision